presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -75,3 +81,28 @@ def __validate_position_over_text(self, analyzer_result, text_len):
                 f"Invalid analyzer result: '{analyzer_result}', "
                 f"original text length is only {text_len}."
             )
+
+    def __extract_anonymizer(self, transformation):
+        anonymizer = transformation.get("anonymizer")()
+        # if the anonymizer is not valid, a InvalidParamException
+        anonymizer.validate(params=transformation)
+        return anonymizer
+
+    def __anonymize_section(
+        self, anonymizer, output_text, analyzer_result, transformation
+    ):
+        text_to_anonymize = output_text[analyzer_result.start : analyzer_result.end]
+        anonymized_text = anonymizer.anonymize(
+            params=transformation, text=text_to_anonymize
+        )  # TODO: [ADO-2754] replace with the singleton class instance
```

Comment:
There was a huge conversation about this one...
including 
@omri374 @itye-msft @SharonHart and myself. I think this should be decided.

Reply:
removed the comment - the decision was not to use singletons. But we need to make sure not to re-create the anonymizer on each request